File: BE/models/Product.js
const mongoose = require('mongoose')
const productDetails = new mongoose.Schema({
    name: String,
    price: Number,
    rating: Number,
    amountOfBuys: Number,
    description: String,
    comments: [String],
    category: {
        type: String,
        enum: ['עגילים', 'צמידים', 'שעונים', 'שרשראות', 'טבעות']
    },
    color: {
        type: String,
        enum: ['כסף', 'זהב']
    },
    sales: [{
        type: mongoose.Schema.Types.ObjectId,
        ref: 'Sale'
    }],
    imageUrl: String,
    dateOfStart: Date,
})
module.exports = mongoose.model('Product', productDetails)

File: BE/modals/Product.py
from mongoengine import (
    Document, StringField, FloatField, IntField, ListField,
    ReferenceField, DateTimeField
)

class Product(Document):
    name = StringField()
    price = FloatField()
    rating = FloatField()
    amountOfBuys = IntField()
    description = StringField()
    comments = ListField(StringField())
    category = StringField(choices=['עגילים', 'צמידים', 'שעונים', 'שרשראות', 'טבעות'])
    color = StringField(choices=['כסף', 'זהב'])
    sales = ListField(ReferenceField('Sale'))
    imageUrl = StringField()
    dateOfStart = DateTimeField()
    views = IntField()